### PR TITLE
CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by Lunarmagpie**
This is a rough draft. maintianers feel free to change anything, but we have to agree on it before its merged.

### CODE REVIEW

File: CONTRIBUTING.md
Comment (trag1c) [inline CONTRIBUTING.md:7]:
```suggestion
mCodingBot bot uses poetry for package managing. Poetry can be installed [with their official instructions](https://python-poetry.org/docs/#installation).
```

File: CONTRIBUTING.md
Comment (trag1c) [inline CONTRIBUTING.md:9]:
```suggestion
Before you start coding, you have to activate your env with `poetry shell`.
```

File: CONTRIBUTING.md
Comment (trag1c) [inline CONTRIBUTING.md:15]:
```suggestion
These commands can be run to lint your code:
```

File: CONTRIBUTING.md
Comment (trag1c) [inline CONTRIBUTING.md:37]:
```suggestion
Code from the bot itsel is always imported as `from bot.a import b`.
```

**Review by trag1c — CHANGES_REQUESTED** [review]

**Comment by Lunarmagpie** [comment]
Waiting for @CircuitSacul to set this to the version of Contributing.md on his markdown website thing

**Comment by CircuitSacul** [comment]
[CONTRIBUTING.md](https://github.com/mcb-dev/mCodingBot/files/9641605/CONTRIBUTING.md)

**Comment by Lunarmagpie** [comment]
Is this still a draft?

File: CONTRIBUTING.md
Comment (Lunarmagpie) [inline CONTRIBUTING.md:37]: I don't think "open the source" makes sense. I think "repository", "project", or "source code" are better options.

File: CONTRIBUTING.md
Comment (qexat) [inline CONTRIBUTING.md:23]: Isn't it `PostgreSQL`...?
```suggestion
First, if you're making changes that require the database, you will need to set up PostgreSQL. Once installed, create a user named `mcodingbot`, and then a database called `mcodingbot` that is owned by that user.
```

File: CONTRIBUTING.md
Comment (qexat) [inline CONTRIBUTING.md:25]:
```suggestion
To run the bot, you first need to set up your `config.json` file. Look at [config.json.example](https://github.com/mcb-dev/mCodingBot/blob/main/config.json.example) for an example configuration. Make sure to set `no_db_mode: true` if you chose not to set up Postgres. If you did set up Postgres, you only need to put your password (if any) in the `config.json`.
```
I guess?

File: CONTRIBUTING.md
Comment (qexat) [inline CONTRIBUTING.md:27]: Out of scope, but wouldn't it be simpler to add the script in the `[tools.poetry.scripts]` section of the `pyproject.toml` file?
Then it could be run by simply doing `poetry run mcodingbot`
In that case this could be the subject of another PR, but I'd be interested in your thoughts on this.

File: CONTRIBUTING.md
Comment (qexat) [inline CONTRIBUTING.md:31]:
```suggestion
To create a pull request to mCodingBot, first create a new branch for your changes.
```
Well played lol but I read code attentively

**Review by qexat — CHANGES_REQUESTED** [review]
Some aren't actual request changes but one is kind of mandatory, filthy troller :trollface:

File: CONTRIBUTING.md
Comment (qexat) [reply CONTRIBUTING.md:27]: Created a new issue (73) so you can comment there

File: CONTRIBUTING.md
Comment (qexat) [reply CONTRIBUTING.md:23]: > Both names are in use :v
> 
> From [Wikipedia](https://en.wikipedia.org/wiki/PostgreSQL):
> 
> > _PostgreSQL, also known as Postgres, [...]._

oops, my bad

Comment (Lunarmagpie) [reply CONTRIBUTING.md:23]: I prefer postgres here. It's shorter.

Comment (qexat) [reply CONTRIBUTING.md:23]: > I prefer postgres here. It's shorter.

lol it's shorter by two chars though...

Comment (Lunarmagpie) [reply CONTRIBUTING.md:23]: It's also shorter by two syllables

File: CONTRIBUTING.md
Comment (trag1c) [inline CONTRIBUTING.md:27]:
```suggestion
Once the config is set up, you can run the bot with `poetry run python3 -m mcodingbot`.
```

File: CONTRIBUTING.md
Comment (trag1c) [inline CONTRIBUTING.md:31]:
```suggestion
To create a pull request to mCodingBall, first, create a new branch for your changes.
```

**Review by trag1c — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md
Comment (Lunarmagpie) [reply CONTRIBUTING.md:25]: +1

File: CONTRIBUTING.md
Comment (Lunarmagpie) [reply CONTRIBUTING.md:27]: Most definitely out of scope so im resolving this

File: CONTRIBUTING.md
Comment (Lunarmagpie) [reply CONTRIBUTING.md:43]: I find this harder to read. The short lines make you look really far over to see what they do.

File: CONTRIBUTING.md
Comment (Lunarmagpie) [reply CONTRIBUTING.md]: This feels wrong to me because it makes "first" look like a parenthetical. Imo the best thing to do here is reword the entire sentence if you aren't happy with it.

**Comment by qexat** [comment]
How is going this PR

**Comment by qexat** [comment]
Bump

File: CONTRIBUTING.md
Comment (trag1c) [reply CONTRIBUTING.md:43]: > I find this harder to read. The short lines make you look really far over to see what they do.

idk for me the current one is more cluttered and harder to read

**Comment by MithicSpirit** [comment]
Please make all lines at most 80 characters long so I can read this on a
standard ISO/ANSI screen. There is a reason markdown treats single newlines
as spaces rather than paragraph breaks.

**Comment by qexat** [comment]
bump

**Comment by Lunarmagpie** [comment]
I am ok with merging in the current state, want the other maintainers to review tho.

**Review by MithicSpirit — CHANGES_REQUESTED** [review]
Please make all lines at most 80 characters long so I can read this on a
standard ISO/ANSI screen. There is a reason markdown treats single newlines
as spaces rather than paragraph breaks.

**Comment by Lunarmagpie** [comment]
Honestly I can not be bothered to fix the formatting. If you want it @MithicSpirit then pr/push it into my branch.

**Comment by MithicSpirit** [comment]
Vim can word wrap with `gq` or `gw`, but I can also do it later when I have time.

**Review by trag1c — APPROVED** [review]

**Comment by Lunarmagpie** [comment]
Just changed the max line length manually

File: CONTRIBUTING.md
Comment (Endercheif) [inline CONTRIBUTING.md:54]:
```suggestion
poetry run nox -s lint        # checks if flake8, black, and isort all pass
```

**Review by Endercheif — CHANGES_REQUESTED** [review]

**Review by Endercheif — COMMENTED** [review]
Should code blocks specify their language as `bash` or smth?

**Comment by trag1c** [comment]
> Should code blocks specify their language as `bash` or smth?

No need, I added it for the nox one specifically so that comments are highlighted properly

**Comment by qexat** [comment]
will this be ever merged

**Comment by Lunarmagpie** [comment]
Merging is blocked until mithic gives a positive review

**Comment by trag1c** [comment]
yep, waiting for @MithicSpirit

**Comment by MithicSpirit** [comment]
Waiting on https://github.com/Lunarmagpie/mCodingBotRewrite/pull/1 to give positive review.

**Comment by Lunarmagpie** [comment]
Didn't see that PR sorry. Merged now.

**Review by MithicSpirit — APPROVED** [review]
lgtm now

**Pull request merged**
